doc/source/doc-style/docstrings.rst

```diff
@@ -227,6 +227,32 @@ consider the following function:
 If the definition of the function is updated, this
 section must be updated too.
 
+Type Hints
+==========
+
+By default, Sphinx renders `type hints <https://peps.python.org/pep-0484/>`_ as part
+of the function signature. This can become difficult to read, because the signature
+becomes very long.
+
+Instead, we can render the type hints as part of each parameter's description. For
+this, the ``sphinx.ext.autodoc.typehints``, ``sphinx.ext.napoleon``, and
+``numpydoc`` extensions need to be combined as follows (in ``conf.py``):
+
+.. code:: python
+
+  extensions = [
+      ...
+      "sphinx.ext.autodoc.typehints",
+      "sphinx.ext.napoleon",
+      "numpydoc",
+      ...
+  ]
+  autodoc_typehints = "description"
+
+Note that the order in which the extensions are included matters.
+
```

Comment:
```suggestion
```

Reply:
We might consider keeping this and having this within a `note` directive:

```suggestion
.. note::
   The order in which the extensions are included matters.

```

Reply:
@akaszynski Keeping it is OK with me--I just changed the above to say that had to appear in the following order so viewed the note as redundant. However, it emphasizes the fact, which may be your intent.

Reply:
Ah, seems I messed this up by prematurely accepting the edits. I'll revert it tomorrow, when I'm in front of my usual PC.

@PipKat if we keep the note, would you lean towards keeping the edited version of the sentence above, or revert it to the original?

Reply:
The note directive will definitely put more emphasis on this information. If calling it out is important, use the note!

Reply:
Ok, yeah I think we should use the note. In that case should we revert to the original 
```
Instead, we can render the type hints as part of each parameter's description. For
this, the ``sphinx.ext.autodoc.typehints``, ``sphinx.ext.napoleon``, and
``numpydoc`` extensions need to be combined as follows (in ``conf.py``):
```
or use 
```
Instead, we can render the type hints as part of each parameter's description. To
accomplish this, the ``sphinx.ext.autodoc.typehints``, ``sphinx.ext.napoleon``, and
``numpydoc`` extensions must be combined in the ``conf.py`` file in this order:
```

To me, the latter version now looks somewhat redundant, but wanted to hear your opinion on that.

Reply:
I'd do the last--remove "in this order" from the suggested edit.